service/controller/app/app.go

```diff
@@ -50,9 +53,20 @@ func NewApp(config Config) (*App, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.ImageRegistry must not be empty", config)
 	}
 
-	var resourceSetV1 *controller.ResourceSet
+	// TODO: Remove usage of deprecated controller context.
+	//
+	//	https://github.com/giantswarm/giantswarm/issues/12324
+	//
```

Comment:
I created a TODO so we can do this later. As the Helm v3.2.4 changes are already big.